### PR TITLE
chore(compass-settings): update settings list and input component to have more granular state subscription COMPASS-7098

### PR DESCRIPTION
**Pull request by gribnoysup**
Another clean-up PR I'm opening separately from the actual feature to make it easier to review.

I wanted to make sure that state subscriptions in the settings form are set up in a way that doesn't cause unnecessary re-renders when any value is changed in the store, so this patch changes settings-list to make every settings input a connected component that picks up required primitive values from the store based on `name` instead of settings list mapState functions generating new objects every render whether or not they changed. This also aligns the implementation with [redux style guide](https://redux.js.org/style-guide/#connect-more-components-to-read-data-from-the-store).

While doing this change I stumbled on a few existing small issues with the reducer: un-synced input value causing weird component behavior (which was a bit obscured by the tests because they were not testing the components with a store and actions connected and so weren't taking the asynchronous behavior of `changeFieldValue` into account, another reason to follow [redux recommendations on testing](https://redux.js.org/usage/writing-tests#guiding-principles) while we are using redux for state management) and unhandled rejection during the field update flow.

### CODE REVIEW

File: configs/mocha-config-compass/main-process.js [hunk 1]
```diff
@@ -5,3 +5,13 @@ app.on('web-contents-created', function (_, webContents) {
...
+// Every compass plugin depends on compass-preferences-model, for a lot of them
+// running tests in electron environments are either spamming "no handler
+// registered" warnings or just not working when expected settings are missing
+// or can't be updated. To handle that we are setting up preferences for every
+// test environment, but making sure that it's in sandbox mode so that nothing
+// is actually written to the disk when preferences change
+process.env.COMPASS_TEST_USE_PREFERENCES_SANDBOX =
+  process.env.COMPASS_TEST_USE_PREFERENCES_SANDBOX ?? 'true';
+// NB: Not adding this as a dep in package.json to avoid circular dependency
+require('compass-preferences-model').setupPreferences();
```
Comment (gribnoysup) [inline configs/mocha-config-compass/main-process.js:17]: Manually tried running test-electron for all packages in the monorepo and except for one test case that I adjusted for this change, everything is passing

File: packages/compass-query-bar/src/components/query-ai.spec.tsx [hunk 2]
```diff
@@ -74,8 +81,18 @@ describe('QueryAI Component', function () {
...
+    beforeEach(async function () {
       store = renderQueryAI();
+      trackUsageStatistics =
+        preferencesAccess.getPreferences().trackUsageStatistics;
+      // 'compass:track' will only emit if tracking is enabled
+      await preferencesAccess.savePreferences({ trackUsageStatistics: true });
+    });
+
+    afterEach(async function () {
+      await preferencesAccess.savePreferences({ trackUsageStatistics });
```
Comment (Anemy) [inline packages/compass-query-bar/src/components/query-ai.spec.tsx:95]: Thanks for this drive by fix!

**Review by Anemy — APPROVED** [review]
lgtm! Nice testing improvements, big fan of the redux practice of having the tests do more integration like behavior with the `Provider`.

**Comment by Anemy** [comment]
Merging this re https://mongodb.slack.com/archives/C05AH835WQN/p1692194250538449?thread_ts=1692191252.732439&cid=C05AH835WQN

**Comment by gribnoysup** [comment]
Thank you!

**Pull request merged**